sharktank/sharktank/utils/hf.py

```diff
@@ -80,3 +127,79 @@ def import_hf_dataset_from_hub(
         param_paths=param_paths,
         output_irpa_file=output_irpa_file,
     )
+
+
+_llama3_hf_to_sharktank_tensor_name_map: dict[str, str] = {
+    "model.embed_tokens.weight": "token_embd.weight",
+    "lm_head.weight": "output.weight",
+    "model.norm.weight": "output_norm.weight",
+    "model.layers.{layer_idx}.input_layernorm.weight": "blk.{layer_idx}.attn_norm.weight",
+    "model.layers.{layer_idx}.mlp.down_proj.weight": "blk.{layer_idx}.ffn_down.weight",
+    "model.layers.{layer_idx}.mlp.gate_proj.weight": "blk.{layer_idx}.ffn_gate.weight",
+    "model.layers.{layer_idx}.mlp.up_proj.weight": "blk.{layer_idx}.ffn_up.weight",
+    "model.layers.{layer_idx}.post_attention_layernorm.weight": "blk.{layer_idx}.ffn_norm.weight",
+    "model.layers.{layer_idx}.self_attn.k_proj.weight": "blk.{layer_idx}.attn_k.weight",
+    "model.layers.{layer_idx}.self_attn.o_proj.weight": "blk.{layer_idx}.attn_output.weight",
+    "model.layers.{layer_idx}.self_attn.q_proj.weight": "blk.{layer_idx}.attn_q.weight",
+    "model.layers.{layer_idx}.self_attn.v_proj.weight": "blk.{layer_idx}.attn_v.weight",
+}
+
+
+def transform_llama3_tensor_name_to_sharktank(
+    tensor: InferenceTensor,
+) -> None | InferenceTensor | Sequence[InferenceTensor]:
+    layer_idx = None
+    layer_idx_pattern = "(model\.layers\.)(\d+)(\..*)"
+    re_match = re.match(layer_idx_pattern, tensor.name)
+    name_to_map = tensor.name
+    if re_match:
+        layer_idx = re_match.group(2)
+        name_to_map = re.sub(layer_idx_pattern, r"\1{layer_idx}\3", tensor.name)
+
+    sharktank_name = _llama3_hf_to_sharktank_tensor_name_map[name_to_map]
+    sharktank_name = sharktank_name.format(layer_idx=layer_idx)
+    return DefaultPrimitiveTensor(data=tensor.as_torch(), name=sharktank_name)
+
+
+def llama3_hf_config_to_sharktank(hf_config: dict[str, Any]) -> dict[str, Any]:
```

Comment:
This functionality should either be in `sharktank/sharktank/layers/configs/llm_configs.py`, I think.

I would like to prevent duplication of this code; if they get out of sync, it will be really annoying to maintain/debug.

Reply:
I moved the config related stuff there.